### PR TITLE
Support bool fields

### PR DESCRIPTION
**Pull request by MaXal**
In the end, I managed to implement private build filtering without a boolean field.
But my first attempt with the boolean field in DB required a lot of rework that might be valuable later on.
So I decided to create a PR. Feel free to decline if it looks unnecessary or if there is an easier way to support boolean values properly.

### CODE REVIEW

File: dashboard/shared/src/components/placeholder.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export function usePlaceholder(
     if ((values == null || values.length === 0)) {
       const value = selected()
       if (value != null && value.length != 0) {
-        return Array.isArray(value) ? value.join(", ") : value
+        return Array.isArray(value) ? value.join(", ") : String(value)
```
Comment (develar) [inline dashboard/shared/src/components/placeholder.ts:15]: `String(value)` -> `value.toString()`

File: pkg/data-query/resultJsonEncoder.go [hunk 2]
```diff
@@ -48,6 +48,20 @@ func writeResult(result *proto.Results, columnNameToIndex map[string]int, column
...
         buffer.B = strconv.AppendUint(buffer.B, uint64(v), 10)
       }
+
+    case *proto.ColBool:
+      for i, v := range *data {
+        if i != 0 {
+          _ = buffer.WriteByte(',')
+        }
+        buffer.B = strconv.AppendBool(buffer.B, v)
+        //if v {
```
Comment (develar) [inline pkg/data-query/resultJsonEncoder.go:58]: please remove commented out code.

File: dashboard/shared/src/PersistentStateManager.ts [hunk 3]
```diff
@@ -104,3 +108,9 @@ export class PersistentStateManager {
     }
   }
 }
+
+function boolFromString(s: string|null) {
+  if (s == "true") return true
```
Comment (develar) [inline dashboard/shared/src/PersistentStateManager.ts:113]: not sure that it is a safe transformation. user string maybe also `true`.

**Review by develar — CHANGES_REQUESTED** [review]

File: dashboard/shared/src/PersistentStateManager.ts (hunk 3 above)
Comment (MaXal) [reply dashboard/shared/src/PersistentStateManager.ts:113]: True, but we can't know from the URL whether it's a string or bool. We can probably encode this somehow in a URL like `b"true"`, but I'm not sure it's worth it.

**Pull request merged**